### ISSUE TITLE
SA1117 treats multi line arguments as only existing on single line.

### ISSUE THREAD

**Issue by strue36** [DotNetAnalyzers/StyleCopAnalyzers#3106]
I came across this behaviour with [SA1117](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1117.md) and wasn't sure if it was correct.

(a) Warning
```csharp
var test = MethodWithThreeParameters(
    "First Line"
    + "Second Line", SecondArgument, ThirdArgument);
```

(b) No warning
```csharp
var test = MethodWithThreeParameters(
    "First Line"
    + "Second Line", SecondArgument,
    ThirdArgument);
```

(c) No warning
```csharp
var test = MethodWithThreeParameters(
    "First Line"
    + "Second Line",
    SecondArgument,
    ThirdArgument);
```

In case (a), the first argument is treated as being on the first line, and presumably warns due to arguments 2 and 3 sharing a line.
In case (b) the first argument is treated as only being on the first line, and the second argument isn't flagged as sharing a line with the first argument.
In case (c) every argument is on its own line and the behaviour is as expected.

Should the first argument be treated as spanning both lines 1 and 2, therefore warning in case (b)?

**Comment by sharwell** [comment]
I believe (b) should definitely be a warning.

**Comment by pantosha** [comment]
@sharwell should warning be reported in case?

```csharp
var test = MethodWithThreeParameters(
    "First Line"
    + "Second Line", SecondArgument,);
```

Right now only functions with 3+ arguments are processed.

**Comment by sharwell** [comment]
It seems like yes, that should be a warning.
